shiny/express/__init__.py

```diff
@@ -1,8 +1,74 @@
 from __future__ import annotations
 
+from ..session import Inputs, Outputs, Session
+from ..session import _utils as _session_utils
+from . import app, layout
 from ._output import output_args, suspend_display
+from ._run import is_express_app, wrap_express_app
+from .display_decorator import display_body
 
 __all__ = (
+    "input",
+    "output",
+    "session",
+    "is_express_app",
     "output_args",
     "suspend_display",
+    "wrap_express_app",
+    "app",
+    "layout",
+    "display_body",
 )
+
+# Add types to help type checkers
+input: Inputs
+output: Outputs
+session: Session
+
+
+# Note that users should use `from shiny.express import input` instead of `from shiny
+# import express` and acces via `express.input`. The former provides a static value for
+# `input`, but the latter is dynamic -- every time `express.input` is accessed, it
+# returns the input for the current session. This will work in the vast majority of
+# cases, but when it fails, it will be very confusing.
+def __getattr__(name: str) -> object:
+    if name == "input":
```

Comment:
Consider detecting if input, output, or session are requested more than once from the same scope. (using `inspect`?)